numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4060,8 +4060,11 @@ initialize_casting_tables(void)
     _npy_can_cast_safely_table[_FROM_NUM][NPY_STRING] = 1;
     _npy_can_cast_safely_table[_FROM_NUM][NPY_UNICODE] = 1;
 
-    /* Allow casts from any integer to the TIMEDELTA type */
-#if @from_isint@ || @from_isuint@
+#if @from_isint@ && NPY_SIZEOF_TIMEDELTA >= _FROM_BSIZE
+    /* Allow casts from smaller or equal signed integers to the TIMEDELTA type */
+    _npy_can_cast_safely_table[_FROM_NUM][NPY_TIMEDELTA] = 1;
+#elif @from_isuint@ && NPY_SIZEOF_TIMEDELTA > _FROM_BSIZE
+    /* Allow casts from smaller unsigned integers to the TIMEDELTA type */
```

Comment:
To clarify, this isn't a value-based assessment--it is based on the size of the type, right?

In other words, `uint32` is automatically allowed to cast to `m8`, but NumPy won't be performing judgements as to whether any given `uint64` would cause problems if it were to be converted to an m8? That's up to the user to check & use `same_kind` casting for, at their own peril?

Reply:
Exactly - this is just emulating the logic used below for comparisons between integer types.